cfme/middleware/provider/__init__.py

```diff
@@ -181,55 +143,60 @@ class Timelines(CFMENavigateStep):
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        mon_btn('Timelines')
+        self.prerequisite_view.toolbar.monitoring.item_select('Timelines')
 
 
 @navigator.register(MiddlewareProvider, 'ProviderServers')
 class ProviderServers(CFMENavigateStep):
     prerequisite = NavigateToSibling('Details')
+    VIEW = ProviderServerAllView
 
     def step(self):
-        sel.click(InfoBlock.element('Relationships', 'Middleware Servers'))
+        self.prerequisite_view.contents.relationships.click_at('Middleware Servers')
 
 
 @navigator.register(MiddlewareProvider, 'ProviderDatasources')
 class ProviderDatasources(CFMENavigateStep):
+    VIEW = ProviderDatasourceAllView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        sel.click(InfoBlock.element('Relationships', 'Middleware Datasources'))
+        self.prerequisite_view.contents.relationships.click_at('Middleware Datasources')
 
 
 @navigator.register(MiddlewareProvider, 'ProviderDeployments')
 class ProviderDeployments(CFMENavigateStep):
+    VIEW = ProviderDeploymentAllView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        sel.click(InfoBlock.element('Relationships', 'Middleware Deployments'))
+        self.prerequisite_view.contents.relationships.click_at('Middleware Deployments')
 
 
 @navigator.register(MiddlewareProvider, 'ProviderDomains')
 class ProviderDomains(CFMENavigateStep):
+    VIEW = ProviderDomainsAllView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        sel.click(InfoBlock.element('Relationships', 'Middleware Domains'))
+        self.prerequisite_view.contents.relationships.click_at('Middleware Domains')
 
 
 @navigator.register(MiddlewareProvider, 'ProviderMessagings')
 class ProviderMessagings(CFMENavigateStep):
+    VIEW = ProviderMessagingAllView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        sel.click(InfoBlock.element('Relationships', 'Middleware Messagings'))
+        self.prerequisite_view.contents.relationships.click_at('Middleware Messagings')
 
 
 @navigator.register(MiddlewareProvider, 'TopologyFromDetails')
 class TopologyFromDetails(CFMENavigateStep):
     prerequisite = NavigateToSibling('Details')
```

Comment:
I don't see view statement here

Reply:
Topology is not Middleware Specific, can we leave it for later?

Reply:
sure, please add appropriate comment there